src/OpenTelemetry/Metrics/ExponentialHistogramBuckets.cs

```diff
@@ -81,7 +86,7 @@ internal Enumerator(long[] buckets)
         /// collection.</returns>
         public bool MoveNext()
```

Comment:
I think the unit tests should also cover a scale-down scenario when the CircularBufferBuckets offset is updated.

Reply:
It's not very obvious, but the existing tests actually do cover this scenario.

Even this simple test that only records two values requires a scale down. When the first value is recorded (10) the initial scale of 20 is fine, but when the second value is recorded (5) a scale down is necessary to fit this range of values within the default 160 buckets.

https://github.com/open-telemetry/opentelemetry-dotnet/blob/d924663dc3d6bfdf737bc49ceaa1addcec90a2d6/test/OpenTelemetry.Tests/Metrics/MetricSnapshotTests.cs#L238-L241

Reply:
I'm going to merge this PR, but I will be coming back to the tests. Clearly there's improvements to make as the bug this PR fixes is not really covered by any tests yet.